react/GalleryLayout.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ComponentType, useContext, useMemo } from 'react'
+import classNames from 'classnames'
+
+import { ProductListContext } from 'vtex.product-list-context'
+import { Spinner } from 'vtex.styleguide'
+import { useCssHandles } from 'vtex.css-handles'
+import {
+  useResponsiveValue,
+  MaybeResponsiveInput,
+} from 'vtex.responsive-values'
+import { useRuntime } from 'vtex.render-runtime'
+import { useSearchPageState } from 'vtex.search-page-context/SearchPageContext'
+
+import GalleryLayoutRow from './components/GalleryLayoutRow'
+import SettingsContext from './components/SettingsContext'
+import ProductListEventCaller from './utils/ProductListEventCaller'
+
+export interface LayoutDescription {
+  name: string
+  component: string
+  itemsPerRow: MaybeResponsiveInput<number>
+}
+
+export type Slots = Record<string, ComponentType>
+
+export interface GalleryLayoutProps {
+  layouts: LayoutDescription[]
+  lazyItemsRemaining: number
+  products: any[]
+  showingFacets: boolean
+  summary: any
```

Comment:
What is this `summary`?

Reply:
It is a set of product-summary props. I didn't know how to type it, especially because the legacy gallery had it as type `any` in propTypes.